esp/src/src-react/components/Metrics.tsx

```diff
@@ -208,20 +238,45 @@ export const Metrics: React.FunctionComponent<MetricsProps> = ({
     }, [metricGraph, metricGraphWidget, options]);
 
     const graphButtons = React.useMemo((): ICommandBarItemProps[] => [
+        {
+            key: "selPrev", title: nlsHPCC.PreviousSelection, iconProps: { iconName: "NavigateBack" },
+            disabled: selectedMetricsPtr < 1 || selectedMetricsPtr >= selectedMetrics.length,
+            onClick: () => {
+                metricGraphWidget.centerOnItem(selectedMetrics[selectedMetricsPtr - 1].name);
+                setSelectedMetricsPtr(selectedMetricsPtr - 1);
+            }
+        },
+        {
+            key: "selNext", title: nlsHPCC.NextSelection, iconProps: { iconName: "NavigateBackMirrored" },
+            disabled: selectedMetricsPtr < 0 || selectedMetricsPtr >= selectedMetrics.length - 1,
+            onClick: () => {
+                metricGraphWidget.centerOnItem(selectedMetrics[selectedMetricsPtr + 1].name);
+                setSelectedMetricsPtr(selectedMetricsPtr + 1);
+            }
+        },
         {
             key: "reset", text: nlsHPCC.Reset, iconProps: { iconName: "Undo" },
             onClick: () => {
                 metricGraphWidget.reset();
-                updateMetricGraph(selectedMetrics);
+                setSelectedSource("");
+                setSelectedMetrics([]);
+                setSelectedMetricsPtr(0);
+                pushUrl(`/workunits/${wuid}/metrics`);
```

Comment:
Does this not push a new copy of the current url onto the history?

Reply:
Sometimes yes - but since the selection state is also stored in the URL (and reset removes that) its probably ok.
I could add a test inside pushUrl to prevent pushing an URL that matches the current one?